client/src/pages/Discovery/Discovery.tsx

```diff
@@ -51,7 +55,8 @@ export default function Discovery(): JSX.Element {
       search: string;
     }>,
   ) => {
-    getAllContestsByAdvanceSearch(search, null, null, null).then((data) => {
+    history.push(`/discovery?search=${search}`);
```

Comment:
Why are we pushing these routes onto the history stack here and below? :)

Reply:
removed discovery, was thinking the path is  in the discovery page